### DISCUSSION TITLE
Could Muse retroactively copyright already forked code?

### DISCUSSION THREAD

**Discussion by SumianVoice** [tenacityteam/tenacity-legacy#712] [General]
I just read an article about the nightmare of changing licensing retrospectively, by _getting past contributors_ to comply with the new agreement. It seems highly illegal, but I had to ask myself the question, is it theoretically possible that Muse might try to muscle out forks of Audacity through legal threats?

I'm wondering if anyone else has a better understanding of how licensing works. I was under the impression that although original creators of material fundamentally own the copyright to that material, if it is ever licensed under GPL for instance, that it could not be re-licensed retrospectively and terminate already licensed copies of the material.

**Comment by leio** [comment]
Copyright holders can license their works how they please - so not so much retroactively, as offering it under a different license. But what is released GPL stays GPL (too), can't retroactively retract it, only potentially stop licensing new work under GPL, at which point forks could have a problem merging in changes into a GPL fork. Same kind of problem between different versions of GPL if there is no "or later" clause, etc. IANAL.
